systems/fib.asm

```diff
@@ -0,0 +1,32 @@
+section .text
+global fib
+fib:
+	; deal with the base case
+	mov eax, edi
+	cmp edi, 1 
+	jle return
+
+	; move callee saved registers to the stack
+	push rbx
```

Comment:
Is there a possibility that the value of `rbx` is random in the first iteration since the move to store `n` into `ebx` happens later in the code?

Reply:
I think so, but it doesn't matter since we never use it in the first iteration